master/buildbot/www/authz/roles.py

```diff
@@ -86,3 +86,18 @@ def getRolesFromUser(self, userDetails):
         if userDetails.get('username') in self.usernames:
             return self.roles
         return []
+
+
+class RolesFromTeams(RolesFromBase):
+
+    def __init__(self, teamPrefix=""):
+        RolesFromBase.__init__(self)
+        self.teamPrefix = teamPrefix
+
+    def getRolesFromUser(self, userDetails):
+        roles = []
+        if 'teams' in userDetails:
+            for team in userDetails['teams']:
```

Comment:
you can simplify the two lines:

    for team in userDetails.get('teams', []):